packages/store-indexer/src/postgres/getLogs.ts

```diff
@@ -0,0 +1,69 @@
+import { PgDatabase } from "drizzle-orm/pg-core";
+import { Hex } from "viem";
+import { StorageAdapterLog, SyncFilter } from "@latticexyz/store-sync";
+import { tables } from "@latticexyz/store-sync/postgres";
+import { and, eq, or } from "drizzle-orm";
+import { decodeDynamicField } from "@latticexyz/protocol-parser";
+import { bigIntMax } from "@latticexyz/common/utils";
+
+export async function getLogs(
+  database: PgDatabase<any>,
+  {
+    chainId,
+    address,
+    filters = [],
+  }: {
+    readonly chainId: number;
+    readonly address?: Hex;
+    readonly filters?: readonly SyncFilter[];
+  }
+): Promise<{ blockNumber: bigint; logs: (StorageAdapterLog & { eventName: "Store_SetRecord" })[] }> {
+  const conditions = filters.length
+    ? filters.map((filter) =>
+        and(
+          address != null ? eq(tables.recordsTable.address, address) : undefined,
+          eq(tables.recordsTable.tableId, filter.tableId),
+          filter.key0 != null ? eq(tables.recordsTable.key0, filter.key0) : undefined,
+          filter.key1 != null ? eq(tables.recordsTable.key1, filter.key1) : undefined
+        )
+      )
+    : address != null
+    ? [eq(tables.recordsTable.address, address)]
+    : [];
+
+  const chainState = await database
+    .select()
+    .from(tables.chainTable)
+    .where(eq(tables.chainTable.chainId, chainId))
+    .execute()
+    .then((rows) => rows.find(() => true));
+  let blockNumber = chainState?.lastUpdatedBlockNumber ?? 0n;
+
+  const records = await database
+    .select()
+    .from(tables.recordsTable)
+    .where(or(...conditions));
+  blockNumber = bigIntMax(
+    blockNumber,
+    records.reduce((max, record) => bigIntMax(max, record.lastUpdatedBlockNumber ?? 0n), 0n)
+  );
+
+  const logs = records
+    .filter((record) => !record.isDeleted)
```

Comment:
should we just include `!isDeleted` in the sql query?

Reply:
I intentionally didn't for a few reasons:
- on the whole, I expect the number of actual deleted records to be low, because it costs gas to delete vs leaving a record laying around
- doing it at the query level will change the query planning and I want to make sure we hit the indexes I've set up here (not to say that we _can't_ but I wanna make sure we can make use of the indexes)
- we'll be doing similar shaped queries for the update stream (query all records and filter or return whether a record is deleted) and seems wise to use the same SQL query path for caching

Reply:
> on the whole, I expect the number of actual deleted records to be low, because it costs gas to delete vs leaving a record laying around

Independent of the number of deleted records it would save us one linear operation over the entire logs array if we don't have to filter for `isDeleted` after fetching all the logs. If we'd add another index for `isDeleted` we could avoid that linear operation, at the cost of more expensive inserts (but since inserts are throttled by the blockchain, write performance is less critical than read performance)

> we'll be doing similar shaped queries for the update stream (query all records and filter or return whether a record is deleted) and seems wise to use the same SQL query path for caching

For the update stream the query would also include the latest block number no? So wouldn't it already be a slightly different query shape?

Reply:
> For the update stream the query would also include the latest block number no? So wouldn't it already be a slightly different query shape?

ah yeah, that's true

mind if we come back to this? it would be nice to set up some scaffolding for testing/measuring query performance, processing performance (SQL query -> HTTP response), etc. and I don't want to blow up this PR

Reply:
yeah definitely, this is not blocking, approved the PR https://github.com/latticexyz/mud/pull/1965#pullrequestreview-1754966405